Content.Server/Cargo/Systems/CargoSystem.Shuttle.cs

```diff
@@ -190,7 +193,17 @@ private List<(EntityUid Entity, CargoPalletComponent Component, TransformCompone
                 continue;
             }
 
+            if (sellOnly && comp.PalletType != "sell" && comp.PalletType != "both")
+            {
+                continue;
+            }
+            if (buyOnly && comp.PalletType != "buy" && comp.PalletType != "both")
+            {
+                continue;
+            }
+
```

Comment:
And down here you can just do something like

```C#
if (palletType != comp.PalletType && comp.PalletType != PalletType.Both)
   continue;
```
Which will cover both cases.

(I'm writing this on my phone, so you may need to modify things a bit to make it work)